index.js

```diff
@@ -94,20 +148,48 @@ class FSMerge {
     return result;
   }
 
-  readDirSync(dirPath, options) {
+  readdirSync(dirPath, options) {
     let { _dirList } = this;
-    let result = [];
+    let result = [], errorCount = 0;
+    let fullDirPath = '';
     for (let i=0; i < _dirList.length; i++) {
       let { root } = getRootAndPrefix(_dirList[i]);
-      let fullDirPath = root + '/' + dirPath;
+      fullDirPath = root + '/' + dirPath;
       fullDirPath = fullDirPath.replace(/(\/|\/\/)$/, '');
       if(fs.existsSync(fullDirPath)) {
         result.push.apply(result, fs.readdirSync(fullDirPath, options));
+      } else {
+        errorCount += 1;
       }
     }
+    if (errorCount == _dirList.length) {
+      fs.readdirSync(fullDirPath);
+    }
     return [...new Set(result)];
   }
 
+  readdir(dirPath, callback) {
+    let result = [];
+    let promise = new Promise((res, rej) => {
+      try {
+        result = this.readdirSync(dirPath);
+      } catch(err) {
+        res({
+          err: err
+        });
+        return;
+      }
+      res({
+        err: null,
+        list: result
+      });
+      return;
+    });
+    promise.then((result) => {
+      callback(result.err, result.list && [...new Set(result.list)]);
```

Comment:
If this callback throws, the promises will be rejected in a unexpected way, we should mimic how the normal async fs callbacks handle this.

Reply:
The simplest would be something like:

```js
let result, error;
try {
   result = this.readdirSync(dirPath);
 } catch (err) {
   error = err;
}

setTimeout(() => callback(error, result), 0);
```